### PR TITLE
Feature/user story nav bar

### PR DESCRIPTION
**Pull request by krutzfeld**
feature navbar

### CODE REVIEW

File: index.htm [hunk 1]
```diff
@@ -1,160 +1,59 @@
...
-<h2>We Make Your Home Alive</h2>
+  <nav>
+    <ul id="nav-links">
+      <li><a href="./index.htm">Home</a></li>
+      <li><a href="./sites/about-us.htm">Blog</a></li>
+      <li><a href="./sites/venus-flytrap-details.htm">Location</a></li>
+      <li><a href="./sites/begonia-details.htm">Shop</a></li>
+    </ul>
+    <input type="search" id="search-box" name="search" placeholder="Search...">
+    <a href="#" class="button outlined-button">Login</a>
```
Comment (krutzfeld) [inline index.htm:25]: 👍  nice split of style logic "button and outlined button"

File: index.htm [hunk 2]
```diff
@@ -1,160 +1,59 @@
...
-<div id="divLeft">
-<a href="./sites/sunflower-details.htm" target="_self">
-  <img src="./res/img/sunflower.jpg" alt="Sunflower" class="imgDetails">
-</a>
-</div>
-<div id="divMiddle">
-<b>Sunflower</b><br>
-Helianthus
-</div>
+  <div id="divRight">
```
Comment (krutzfeld) [inline index.htm]: - unused id
- make sure to use kebap-case in CSS: https://www.theserverside.com/definition/Kebab-case
- its a duplicate id

File: index.htm [hunk 3]
```diff
@@ -1,160 +1,59 @@
...
+  Maintenance Difficulty:  Hard
+  </div> 
+  <div id="divLeft">
```
Comment (krutzfeld) [inline index.htm]: see above

File: index.htm [hunk 4]
```diff
@@ -1,160 +1,59 @@
...
+  <a href="./sites/venus-flytrap-details.htm" target="_self">
+    <img src="./res/img/venus-flytrap.jpg" alt="Venus-Flytrap Flower" class="imgDetails">
+  </a>
+  </div>
+  <div id="divMiddle">
```
Comment (krutzfeld) [inline index.htm]: see above

File: index.htm [hunk 5]
```diff
@@ -1,160 +1,59 @@
...
+  <b>Venus-Flytrap</b><br>
+  Dionaea Miscipula
+  </div>
+
+  <div id="divRight">
```
Comment (krutzfeld) [inline index.htm]: see above

File: index.htm [hunk 6]
```diff
@@ -1,160 +1,59 @@
...
+  Maintenance Difficulty:  Easy
+  </div> 
+  <div id="divLeft">
```
Comment (krutzfeld) [inline index.htm]: see above

File: index.htm [hunk 7]
```diff
@@ -1,160 +1,59 @@
...
+  <a href="./sites/sunflower-details.htm" target="_self">
+    <img src="./res/img/sunflower.jpg" alt="Sunflower" class="imgDetails">
+  </a>
+  </div>
+  <div id="divMiddle">
+    <b>Sunflower</b><br>
+    Helianthus
+  </div>
 
 <div id="divRight">
```
Comment (krutzfeld) [inline index.htm]: see above

File: styles.css [hunk 8]
```diff
@@ -0,0 +1,102 @@
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@500&display=swap');
+
+* {
+  margin: 0;
```
Comment (krutzfeld) [inline styles.css]: margin: 0 is unnecessary code because it is the default. Why did you add this? (Did you try to overwrite something?) 

Also read this discussion on unused code: https://stackoverflow.com/questions/15699995/could-someone-explain-the-pros-of-deleting-or-keeping-unused-code#:~:text=Unused%20code%20is%20a%20bigger,processes%20and%20adds%20significant%20noise.

File: styles.css [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+  padding: 0;
+  box-sizing: border-box;
+}
+
+body {
+  font-family: "Montserrat", Arial, Helvetica;
```
Comment (krutzfeld) [inline styles.css]: Add a generic font as a fallback option

File: styles.css [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+li, a {
+  font-weight: 500;
```
Comment (krutzfeld) [inline styles.css]: font-weight has no effect

File: styles.css [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+  font-size: 12px;
```
Comment (krutzfeld) [inline styles.css]: set font-sizes in body as a default and then if you want to have larger font define classes like "text-sm", "text-l", "text-xl" with font-sizes accordingly.

Imagine you have to refactor what is a large text into 16px for example. With this approach you would have to crawl through all classes and definitions and update it. Ooof.

File: styles.css [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+  text-decoration: none;
```
Comment (krutzfeld) [inline styles.css:12]: dont set the initial values explicitly

File: styles.css [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+  border: 1px solid rgb(0, 110, 82);
+  padding: 7px 10px 7px 10px;
+  text-align: center;
+  text-decoration: none;
+  display: inline-block;
+  margin: 8px 4px;
+  transition-duration: 0.2s;
+  cursor: pointer;
+  border-radius: 4px;
+  justify-content: flex-end;
```
Comment (krutzfeld) [inline styles.css]: `justify-content: flex-end;` seems to have no effect

File: styles.css [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+  color: white;
+}
+
+.filled-button:hover {
+  background-color: #212529;
+  color: white;
+ }
+
+#search-box {
+  width: 150px;
```
Comment (krutzfeld) [inline styles.css:69]: I would be careful with specific widths.
Why? Look at the screenshot. If you have a small device the browser has no choice but to give the search field 150px. That might be half of the nav-bard width.

Find a solution that works with resizing

Screen Shot 2022-04-19 at 11 23 51

File: styles.css [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+  text-decoration: none;
+  color: white;
+}
+
+nav {
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  background-color:#212529;
+  padding: 3px 15%;
```
Comment (krutzfeld) [inline styles.css]: This gives it a padding of 3px on top, and 15% on the right. This is not really feasible. I know you want a placeholder left and right. But for this you can use a flex spacer that scales better.

Look at the problem here:
image

File: styles.css [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+  background-color: #212529;
+  color: white;
+ }
+
+#search-box {
+  width: 150px;
+  border-radius: 4px;
+  padding: 7px 7px 7px 7px;  
+  margin: 8px 4px;
+  justify-content: flex-end;
```
Comment (krutzfeld) [inline styles.css]: what effect does justify-content have here?

File: styles.css [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+
+.filled-button:hover {
+  background-color: #212529;
+  color: white;
+ }
+
+#search-box {
+  width: 150px;
+  border-radius: 4px;
+  padding: 7px 7px 7px 7px;
```
Comment (krutzfeld) [inline styles.css]: you can use padding: 7px for that.

File: styles.css [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+  width: 150px;
+  border-radius: 4px;
+  padding: 7px 7px 7px 7px;  
+  margin: 8px 4px;
+  justify-content: flex-end;
+}
+  
+#search-box:focus {
+  border: 4px solid rgb(0, 110, 82);
+  outline: none;
```
Comment (krutzfeld) [inline styles.css]: It seems this is the initial value again?

File: styles.css [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+.wrapper {
+  display: -webkit-box;
```
Comment (krutzfeld) [inline styles.css]: Only display flex is used. Everything else is overwritten hence useless.

If you want to target specific platforms (Mozilla, internet explorer) there are selectors for that, but IMO this is an absolute edge case and you should avoid that at all cost. Make sure that you support the platforms that are required and do not code custom solutions for every platform.

Here you can find what platforms support flex https://caniuse.com/flexbox
What that basically means - if a browser is too old for that it is completely outdated.

Screen Shot 2022-04-19 at 11 44 05

File: styles.css [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+  display: -ms-flexbox;
+  display: -webkit-flex;
+  display: flex;
+  -webkit-flex-flow: row wrap;
+  flex-flow: row wrap;
+  position: relative;
+  top: 50px;
+}
+
+.wrapper > * {
```
Comment (krutzfeld) [inline styles.css]: You want to select everything that is in wrapper? Why?

File: styles.css [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+  padding: 0px;
```
Comment (krutzfeld) [inline styles.css]: this is unneccessary since its default

File: styles.css [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+  flex: 1 100%;
```
Comment (krutzfeld) [inline styles.css:94]: why not giving wrapper flex layout?

File: index.htm [hunk 23]
```diff
@@ -1,160 +1,59 @@
...
-  <form class="searchForm">
-    <input type="text" name="search" placeholder="Search...">
-  </form>
-</div>
-<h1>WaterMi</h1>
-<h2>We Make Your Home Alive</h2>
+  <nav>
+    <ul id="nav-links">
+      <li><a href="./index.htm">Home</a></li>
+      <li><a href="./sites/about-us.htm">Blog</a></li>
```
Comment (krutzfeld) [inline index.htm]: delete href - this way nothing happens, but its not breaking

File: index.htm [hunk 24]
```diff
@@ -1,160 +1,59 @@
...
-  <button class="button2">Sign-up</button>
-  <button class="button1">Login</button>
-  <form class="searchForm">
-    <input type="text" name="search" placeholder="Search...">
-  </form>
-</div>
-<h1>WaterMi</h1>
-<h2>We Make Your Home Alive</h2>
+  <nav>
+    <ul id="nav-links">
```
Comment (krutzfeld) [inline index.htm:18]: You make it a list, which it is not and then remove the appearance via CSS. Why?
It seem you can achieve the same thing with less

```html
    <div id="nav-links">
        <a href="./index.htm">Home</a>
        <a href="./sites/about-us.htm">Blog</a>
        <a href="./sites/venus-flytrap-details.htm">Location</a>
        <a href="./sites/begonia-details.htm">Shop</a>
    </div>
```

and then in CSS

```css
#nav-links {
  margin-right:auto;
}

#nav-links a {
  display: inline-block;
  color: white;
  padding-left: 6px;
}
```

File: index.htm [hunk 25]
```diff
@@ -1,160 +1,59 @@
...
+      <li><a href="./index.htm">Home</a></li>
+      <li><a href="./sites/about-us.htm">Blog</a></li>
+      <li><a href="./sites/venus-flytrap-details.htm">Location</a></li>
+      <li><a href="./sites/begonia-details.htm">Shop</a></li>
+    </ul>
+    <input type="search" id="search-box" name="search" placeholder="Search...">
```
Comment (krutzfeld) [inline index.htm:24]: Look at that closely
Screen Shot 2022-04-19 at 12 00 16

it looks like you have 2 colored borders there, but if you focus the input "jumps" - you should always avoid movement when focusing an element, because the user is looking at the cursor as a point of reference. It's disturbing (or maybe just annoying) if you click smth and it moves away.

File: styles.css [hunk 26]
```diff
@@ -0,0 +1,102 @@
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@500&display=swap');
+
+* {
```
Comment (cpmortley) [inline styles.css:3]: Test

File: styles.css (hunk 8 above)
Comment (cpmortley) [reply styles.css]: When I remove this there is in fact a margin that shows up.

File: styles.css (hunk 9 above)
Comment (cpmortley) [reply styles.css]: Isn't "Arial" and Helvetica generic?

File: styles.css (hunk 11 above)
Comment (cpmortley) [reply styles.css]: If I do this, I will then have to explicitly state the font sizes for H1, H2, H3 and p.  Won't this require more code?

File: styles.css (hunk 12 above)
Comment (cpmortley) [reply styles.css:12]: Can you please clarify here?  I am not understanding this comment.

File: styles.css (hunk 18 above)
Comment (cpmortley) [reply styles.css]: Can you clarify what you mean by this?

File: styles.css (hunk 12 above)
Comment (cpmortley) [reply styles.css:12]: Should I create a class for li and a separate class for a?

File: styles.css (hunk 15 above)
Comment (cpmortley) [reply styles.css]: I have eliminated this problem with "@media screen and (max-width: 600px)".  Is that the solution you are looking for?

File: index.htm (hunk 24 above)
Comment (cpmortley) [reply index.htm:18]: This code does not look familiar to me.  My code currently looks like the screen shot below.  From this code I interpreted the links on the nav bar as a list.  This is: Home, Blog, Location, Shop.  This is a list of items.  I obtained this instruction from the following link in W3 schools:  [https://www.w3schools.com/css/tryit.asp?filename=trycss_navbar_horizontal_black]
Capture

File: styles.css (hunk 22 above)
Comment (cpmortley) [reply styles.css:94]: See above

File: styles.css (hunk 20 above)
Comment (cpmortley) [reply styles.css]: Because if I don't, the page looks like this-->
Capture

File: styles.css (hunk 8 above)
Comment (krutzfeld) [reply styles.css]: You are right. It is better though to add a separate reset.css sheet. I added that.

File: styles.css (hunk 9 above)
Comment (krutzfeld) [reply styles.css]: not really. These are specific (well common) fonts. Here your find the documentation of CSS / HTML:
https://developer.mozilla.org/en/docs/Web/CSS/font-family
Screen Shot 2022-04-22 at 9 56 08

sans-serif is for example generic :)

File: styles.css (hunk 11 above)
Comment (krutzfeld) [reply styles.css]: Not really, you can just leave the h1 elements at it is but not style every li element as 12px in general. Why did you choose 12px for all list and anchor elements in the first place?

what I meant:
```
.text-sm{
  font-size: 8px;
}

.text-l{
  font-size: 16px;
}

.text-xl{
  font-size: 20px;
}
```

```
<li class="text-l">
...
</li>
```

File: styles.css (hunk 12 above)
Comment (krutzfeld) [reply styles.css:12]: No. Just text-decoration: none is the default value, isn't it?

File: styles.css (hunk 14 above)
Comment (krutzfeld) [reply styles.css:69]: I very much like the current solution!
👍

File: styles.css (hunk 21 above)
Comment (krutzfeld) [reply styles.css]: Refers to reset.css

File: index.htm (hunk 24 above)
Comment (krutzfeld) [reply index.htm:18]: Okay. Apparently there is a discussion where the common understanding is like yours :) https://stackoverflow.com/questions/5544885/should-i-use-uls-and-lis-inside-my-navs

Nice ;)

File: index.htm (hunk 25 above)
Comment (krutzfeld) [reply index.htm:24]: it still jumps..

File: styles.css (hunk 22 above)
Comment (krutzfeld) [reply styles.css:94]: it seems you resolved that

File: styles.css (hunk 20 above)
Comment (krutzfeld) [reply styles.css]: 1. you have 2 .wrapper definitions...

2. The mistake you made was declaring the wrapper as a row. As a row means that everything inside is a column - which causes the effect you see. Declaring it as a column (one big column) makes everything a row.

```css
.wrapper {
  display: flex;
  flex-flow: column;
  position: relative;
  top: 50px;
}
```

looks like
image

File: styles.css (hunk 15 above)
Comment (krutzfeld) [reply styles.css]: This was very nice work! 👍

File: styles.css (hunk 18 above)
Comment (krutzfeld) [reply styles.css]: outline: none is the default value of outline.

File: styles.css (hunk 9 above)
Comment (cpmortley) [reply styles.css]: Made the changes with "sans-serif" added.

File: styles.css (hunk 12 above)
Comment (cpmortley) [reply styles.css:12]: If I remove text-decoration the words will be underlined.  I do not want that.

File: styles.css (hunk 18 above)
Comment (cpmortley) [reply styles.css]: If I remove outline: none it does not remove the white outline
Capture

Comment (cpmortley) [reply styles.css]: This has been fixed.

File: index.htm (hunk 25 above)
Comment (cpmortley) [reply index.htm:24]: This has been fixed.

File: styles.css (hunk 20 above)
Comment (cpmortley) [reply styles.css]: This has been fixed.

File: styles.css (hunk 11 above)
Comment (cpmortley) [reply styles.css]: I chose 12px because from my initial estimation of your example I should  be following it looked like 12px.  I know I should have asked font size specifications before beginning.  I will make sure to do this next time.  Now that we are on the topic, can you please specify what font sizes you want for H1, H2, H3, p, li, buttons, inputs and a?

File: index.htm (hunk 25 above)
Comment (krutzfeld) [reply index.htm:24]: Nice!

Comment (krutzfeld) [reply index.htm:24]: Nice!

**Comment by krutzfeld** [comment]
There is an issue that came up - which is the last one it seems...

The top navbar still collides with the wrapper content. You can see that in the element inspector of chrome.
Try to find out why.
